client/src/Components/Notifications/index.js

```diff
@@ -0,0 +1,73 @@
+/* eslint-disable no-unused-vars */
+import { useEffect, useState } from 'react';
+import { Link } from 'react-router-dom';
+
+const Notifications = ({ username }) => {
+	const [notifications, setNotifications] = useState(
+		localStorage.getItem('notifications')
+			? JSON.parse(localStorage.getItem('notifications'))
+			: []
+	);
+
+	// for opening and EventSource with the server
+	useEffect(() => {
+		if (username) {
+			const eventSource = new EventSource(
+				`http://localhost:3001/api/v1/users/${username}/notifications`,
+				{ withCredentials: true }
+			);
+			eventSource.onmessage = (e) => {
+				addNotification('A comment was added to your post', e.data);
+			};
+		}
+		// removed addNotification as this function reference every rerender
+		// eslint-disable-next-line react-hooks/exhaustive-deps
+	}, [username]);
+
+	// for updating localStorage
+	useEffect(() => {
+		localStorage.setItem('notifications', JSON.stringify(notifications));
+	}, [notifications]);
+
+	function addNotification(message, link) {
+		setNotifications((oldArray) => {
+			if (!oldArray.some((e) => e.link === link)) {
+				return [...oldArray, { message, link }];
+			} else {
+				return oldArray;
+			}
+		});
+	}
+
+	function deleteNotification(link) {
+		const newNotifications = notifications.filter((e) => e.link !== link);
+		setNotifications(newNotifications);
```

Comment:
Since you're using the setter which is asynchronous, I'd recommend doing filtering inside the setter.

Reply:
I recently noticed a bug where if you visit a new page sometimes this deletion would not be done from the localStorage as the setter is async. In this case, I need the newNotifications array to set the localStorage synchronously if needed.